React/wbooks/src/redux/auth/reducer.js

```diff
@@ -0,0 +1,91 @@
+import { authActions } from './actions';
+
+/* ------------- Auth reducer ------------- */
+const initialState = {
+  logState: {
+    logging: false,
+    loginFail: false,
+    logout: false,
+    signing: false,
+    signupfail: false
+  }
+};
+
+/* eslint-disable complexity */
+export function reducer(state = initialState, action) {
+  switch (action.type) {
+    case authActions.login: {
+      return {
+        ...state,
+        logState: {
+          logging: true,
+          loggedIn: false,
+          loginFail: false
+        }
+      };
+    }
+    case authActions.loginFailure: {
+      return {
+        ...state,
+        logState: {
+          logging: false,
+          loggedIn: false,
+          loginFail: true
+        }
+      };
+    }
+    case authActions.loginSuccess: {
+      return {
+        ...state,
+        logState: {
+          logging: true,
+          loggedIn: true,
+          loginFail: false
+        }
+      };
+    }
+    case authActions.logout: {
+      return {
+        ...state,
+        logState: {
+          logout: true,
+          signing: false,
+          signupfail: false
```

Comment:
if you are logging out, it means you were logged in. So there's no way signupfail or signing are true. No need to set these two.